spec/requests/disks_spec.rb

```diff
@@ -0,0 +1,56 @@
+RSpec.describe "Disks API" do
+  describe "as a subcollection of VMs" do
+    describe "GET /api/vms/:c_id/disks" do
+      it "can list the snapshots of a VM" do
+        api_basic_authorize(subcollection_action_identifier(:vms, :disks, :read, :get))
+        hw = FactoryBot.create(:hardware)
+        vm = FactoryBot.create(:vm_vmware, :hardware => hw)
+        disk = FactoryBot.create(:disk, :hardware => hw)
+
+        _other_disk = FactoryBot.create(:disk)
+
+        get(api_vm_disks_url(nil, vm))
+
+        expected = {
+          "count"     => 2,
+          "name"      => "disks",
+          "subcount"  => 1,
+          "resources" => [
+            {"href" => api_vm_disk_url(nil, vm, disk)}
+          ]
+        }
+
+        expect(response.parsed_body).to include(expected)
+        expect(response).to have_http_status(:ok)
+      end
+    end
+
+    describe "GET /api/vms/:c_id/disks/:s_id" do
+      it "can show a VM's disk" do
+        api_basic_authorize(subcollection_action_identifier(:vms, :disks, :read, :get))
+        hw = FactoryBot.create(:hardware)
```

Comment:
Isn't it better to put `hw, disk, vm` into `lets' to top and use for all `it` blocks ?